x-pack/plugins/enterprise_search/server/utils/indices_search_result_provider.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from, takeUntil } from 'rxjs';
+
+import {
+  GlobalSearchProviderResult,
+  GlobalSearchResultProvider,
+} from '@kbn/global-search-plugin/server';
+import { i18n } from '@kbn/i18n';
+
+import { ENTERPRISE_SEARCH_CONTENT_PLUGIN } from '../../common/constants';
+
+import { getIndexData } from '../lib/indices/utils/get_index_data';
+
+export function getIndicesSearchResultProvider(): GlobalSearchResultProvider {
+  return {
+    find: ({ term, types, tags }, { aborted$, client, maxResults }) => {
+      if (!client || tags || (types && !types.includes('indices'))) {
+        return from([[]]);
+      }
+      const fetchIndices = async (): Promise<GlobalSearchProviderResult[]> => {
+        const { indexNames } = await getIndexData(client, true, false, term);
```

Comment:
This `getIndexData()` call is ignoring the search term because the 2nd parameter `onlyShowSearchOptimizedIndices=true`, which is why you have to filter out score == 0 in the map below. You should be able to utilize the the ES get call to only return matching indices.

Reply:
The nice thing about getIndexData() is that it expands aliases. I'm not sure based on the requirements if it's desired that these show up in search results or not though - this is a good clarification for product. But I agree with Rodney it doesn't appear like we're correctly pulling this information? 

There isn't a good way of pulling indices using the `_search` API, you'd have to use the [Get index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html) API and use a wildcard around your search term.

Reply:
I *think* only showing search optimized indices (that is: indices that start with 'search') is not what we want to do anyway--we want to show all indices, right? Especially now that we've removed the search- start requirement.

Reply:
Then making this `getIndexData(client, false, false, term);` may be the easiest solution?

Reply:
Probably, yes.